PROShine/Views/BattleView.xaml.cs

```diff
@@ -0,0 +1,415 @@
+                    ExpBar.Width = 200;
+                }
+                else
+                {
+                    _expBarWidth = active.Experience.RatioToNextLevel - 0.5;
+                    ExpBar.Width = _expBarWidth * 3.7;
+                }
+            });
+        }
+
+        public void BattleEnded()
+        {
+            Dispatcher.InvokeAsync(delegate
+            {
+                PROShineLogo.Visibility = Visibility.Visible;
+                UIGrid.Visibility = Visibility.Hidden;
+            });
+        }
+
+        public void ConnectionClosed()
+        {
+            Dispatcher.InvokeAsync(delegate
+            {
+                PROShineLogo.Visibility = Visibility.Visible;
+                UIGrid.Visibility = Visibility.Hidden;
+            });
+        }
+
+        public void ActivePokemonChanged()
+        {
+            Dispatcher.InvokeAsync(delegate
+            {
+                Pokemon active = _bot.Game.Team[_bot.Game.ActiveBattle.SelectedPokemonIndex];
+                _activeHPWidth = active.CurrentHealth;
+                ActiveHealthBar.Width = _activeHPWidth / active.MaxHealth * 200;
+                ActiveHealthBar.Background = new SolidColorBrush(EvaluateGradient(_activeHPWidth / active.MaxHealth));
+                if (active.Experience.CurrentLevel == 100)
+                {
+                    ExpBar.Width = 200;
+                }
+                else
+                {
+                    _expBarWidth = active.Experience.RatioToNextLevel - 0.5;
+                    ExpBar.Width = _expBarWidth * 3.7;
+                }
+            });
+        }
+
+        public void OpponentChanged()
+        {
+            Dispatcher.InvokeAsync(delegate
+            {
+                _opponentHPWidth = _bot.Game.ActiveBattle.CurrentHealth;
+                int maxHealth = _bot.Game.ActiveBattle.OpponentHealth;
+                OpponentHealthBar.Width = _opponentHPWidth / maxHealth * 200;
+                OpponentHealthBar.Background = new SolidColorBrush(EvaluateGradient(_opponentHPWidth / maxHealth));
+            });
+        }
+
+        private void AttackButton_Click(object sender, RoutedEventArgs e)
+        {
+            Pokemon active = _bot.Game.Team[_bot.Game.ActiveBattle.SelectedPokemonIndex];
+            ContextMenu attacks = new ContextMenu();
+            bool hasPP = false;
+
+            for (int i = 0; i < active.Moves.Length; i++)
+            {
+                PokemonMove move = active.Moves[i];
+                if (move == null || string.IsNullOrEmpty(move.Name))
+                    continue;
+                MenuItem menuItem = new MenuItem
+                {
+                    Header = move.Name,
+                    InputGestureText = $"({move.PP})",
+                };
+                menuItem.Tag = i + 1;
+                if (move.CurrentPoints > 0)
+                {
+                    menuItem.Click += Attack_Click;
+                    hasPP = true;
```

Comment:
Changed the hasPP behavior so that will still show the other moves even if none of them has PP. (They'll still all be disabled of course)